Emit a file that lists which packages don't use ///-source-of-truth

ericstj opened the pull request:
Here's a sample that adds a file that catalogs which libraries do not use ///-source-of-truth.

This could be used by the docs process to understand which packages to exclude from ingestion into dotnet-api-docs.

I just dumped a text file with each package on a new line.  Let me know if you'd prefer some different format.

ericstj committed:
Emit a file that lists which packages don't use ///-source-of-truth

ericstj commented on src/libraries/Microsoft.Internal.Runtime.DotNetApiDocs.Transport/src/Microsoft.Internal.Runtime.DotNetApiDocs.Transport.proj:
```diff
@@ -36,6 +36,7 @@
       <TfmSpecificPackageFile Include="@(ReferenceAssemblyFile);
                                        @(CompilerGeneratedXmlDocFile)"
                               PackagePath="ref\$(NetCoreAppCurrent)\" />
+      <NoDocPackages Include="@(ProjectReferenceCopyLocalPathNonPrivate->WithoutMetadataValue('UseCompilerGeneratedDocXmlFile', 'true')->WithMetadataValue('IsPackable', 'true')->'%(PackageId)'->Distinct())" />
```
That would be exceptionally verbose.  You wouldn't do the same to a linq statement...

ViktorHofer reviewed (approved):
Is the DotNetApiDocs. transport package actually still used? I heard mixed signals on that.

ViktorHofer commented on src/libraries/Microsoft.Internal.Runtime.DotNetApiDocs.Transport/src/Microsoft.Internal.Runtime.DotNetApiDocs.Transport.proj:
```diff
@@ -36,6 +36,7 @@
       <TfmSpecificPackageFile Include="@(ReferenceAssemblyFile);
                                        @(CompilerGeneratedXmlDocFile)"
                               PackagePath="ref\$(NetCoreAppCurrent)\" />
+      <NoDocPackages Include="@(ProjectReferenceCopyLocalPathNonPrivate->WithoutMetadataValue('UseCompilerGeneratedDocXmlFile', 'true')->WithMetadataValue('IsPackable', 'true')->'%(PackageId)'->Distinct())" />
```
It's the combination of `UseCompilerGeneratedDocXmlFile=false && IsPackable=true`

ericstj committed:
Use ReferencePath to identify packages without compiler generated docs

We were missing non-copy-local assemblies (EG: framework overlapping) that had packages.
EG:
System.Collections.Immutable
System.Text.Json
etc

gewarren commented on src/libraries/Microsoft.Internal.Runtime.DotNetApiDocs.Transport/src/Microsoft.Internal.Runtime.DotNetApiDocs.Transport.proj:
```diff
@@ -46,6 +47,13 @@
     </ItemGroup>
 
+    <WriteLinesToFile Lines="@(NoDocPackages)"
+                      File="$(OutputPath)NoDocPackages.txt"
```
@ericstj I'm not following...why do I need to know about the reference assemblies in the package?

gewarren replied:
I guess I hadn't got that far. I wasn't initially planning to reference this package - just to unpack it and look at the one file.
